helper/resource/testcase_providers.go

```diff
@@ -21,6 +21,10 @@ func (c TestCase) providerConfig(_ context.Context, skipProviderBlock bool) stri
 	//      it does have a special purpose that wasn't being unit tested prior.
 	for name := range c.Providers {
 		providerBlocks.WriteString(fmt.Sprintf("provider %q {}\n", name))
+
+		requiredProviderBlocks.WriteString(fmt.Sprintf("    %s = {\n", name))
+
+		requiredProviderBlocks.WriteString("    }\n")
```

Comment:
I think we can omit this handling here (at least for now) -- its odd to have empty required_provider configuration and honestly not sure if Terraform will always treat that as valid. 👍